tests/integration/helpers/cluster.py

```diff
@@ -203,7 +203,14 @@ def __init__(self, base_path, name=None, base_config_dir=None, server_bin_path=N
         project_name = pwd.getpwuid(os.getuid()).pw_name + p.basename(self.base_dir) + self.name
         # docker-compose removes everything non-alphanumeric from project names so we do it too.
         self.project_name = re.sub(r'[^a-z0-9]', '', project_name.lower())
-        self.instances_dir = p.join(self.base_dir, '_instances' + ('' if not self.name else '_' + self.name))
+        instances_dir_name = '_instances'
+        if self.name:
+            instances_dir_name += '_' + self.name
+
+        if 'INTEGRATION_TESTS_RUN_ID' in os.environ:
+            instances_dir_name += '_' + os.environ['INTEGRATION_TESTS_RUN_ID']
```

Comment:
Ok.

I'd also make escaping or sanitization when env variable is concatenated with path. Not because it is needed but as a matter of culture.